File: Navtrack.Api.Model/Assets/GetTripsModel.cs
using System.Linq;
using Navtrack.Api.Model.Custom;

namespace Navtrack.Api.Model.Assets
{
    public class GetTripsModel : ResultsModel<TripModel>
    {
        public int TotalDistance => Results.Sum(x => x.Distance);
        public int TotalLocations => Results.Sum(x => x.Locations.Count);
    }
}

File: Navtrack.Core/ClientApp/src/apis/types/asset/GetTripsModel.ts
import { ResultsModel } from "../ResultsModel";
import { TripModel } from "./TripModel";

export interface GetTripsModel extends ResultsModel<TripModel> {
  totalDistance: number;
  totalLocations: number;
}

export const DefaultGetTripsModel: GetTripsModel = {
  totalDistance: 0,
  totalLocations: 0,
  results: []
};
